### PR TITLE
bugfixing in tests

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

**Comment by chfast** [comment]
What is so important about it?

**Comment by gluk256** [comment]
Netwoking tests with JS console don't work at all.

File: libp2p/Network.cpp [hunk 1]
```diff
@@ -208,19 +208,17 @@ bi::tcp::endpoint Network::traverseNAT(std::set<bi::address> const& _ifAddresses
...
 	try
 	{
-		givenPort = stoi(split.at(1));
-	}
-	catch (...)
-	{
-		clog(NetWarn) << "Error resolving host address..." << LogTag::Url << _addr << ". Could not find a port after ':'";
-		return bi::tcp::endpoint();
+		if (split.size() > 1)
+			port = static_cast<unsigned>(stoi(split.at(1)));
```
Comment (chfast) [inline libp2p/Network.cpp]: I don't know how important it is, but negative port number are valid here. E.g. "1.1.1.1:-13".
You can use `std::stoul` to parse port as an unsigned value. Checking for too high values would be good too.

Comment (chfast) [inline libp2p/Network.cpp]: This is a good candidate for a unit test.

**Comment by gavofyork** [comment]
looks fine. but what bugs is it supposed to fix?

File: test/libwhisper/shhrpc.cpp [hunk 2]
```diff
@@ -350,8 +350,15 @@ BOOST_AUTO_TEST_CASE(server)
 	string sess2("session number two");
 	jsonrpcServer->addSession(sess2, permissions);
 	
-	b = jsonrpcServer->admin_web3_setVerbosity(5, sess1);
+	int newVerbosity = 10;
+	int oldVerbosity = g_logVerbosity;
```
Comment (chfast) [inline test/libwhisper/shhrpc.cpp]: That checks nothing if g_logVerbosity is 10.

**Comment by subtly** [comment]
The bug is that passing "127.0.0.1" causes the method to return an empty endpoint. Thus, when trying to connect to "poc-9.ethdev.com" or "127.0.0.1", it will return "0.0.0.0".

File: libdevcore/Log.h [hunk 3]
```diff
@@ -85,7 +85,7 @@ template <class Channel> bool isChannelVisible() { return isChannelVisible(&type
 /// Not thread-safe, use with caution!
 struct VerbosityHolder
 {
-	VerbosityHolder(int _temporaryValue): oldLogVerbosity(g_logVerbosity) { g_logVerbosity = _temporaryValue; }
+	VerbosityHolder(int _temporaryValue, bool _force = false): oldLogVerbosity(g_logVerbosity) { if (g_logVerbosity >= 0 || _force) g_logVerbosity = _temporaryValue; }
```
Comment (chfast) [inline libdevcore/Log.h]: Do you plan to use the _force param? If not, do not add a feature for uncertain future needs.

Comment (gluk256) [inline libdevcore/Log.h]: I need it for local tests.

**Pull request merged**